src/WebJobs.Script/Rpc/RpcInitializationService.cs

```diff
@@ -59,5 +65,30 @@ private async Task InitializeRpcServerAsync()
                 var hostInitEx = new HostInitializationException($"Failed to start Rpc Server. Check if your app is hitting connection limits.", grpcInitEx);
             }
         }
+
+        internal Task InitializeChannelsAsync()
+        {
+            string workerRuntime = _environment.GetEnvironmentVariable(LanguageWorkerConstants.FunctionWorkerRuntimeSettingName);
+            if (_environment.IsLinuxAppServiceEnvironment())
+            {
+                return Task.CompletedTask;
+            }
+            if (_environment.IsLinuxContainerEnvironment())
+            {
+                return Task.CompletedTask;
+            }
+            if (string.IsNullOrEmpty(workerRuntime))
```

Comment:
Add the check for placeholder mode here to avoid launching worker processes for other runtime targets (e.g. launching Java for a .NET function app)